src/utils/DependsOnManager.ts

```diff
@@ -0,0 +1,57 @@
+import { Utils } from './Utils';
+import { $$ } from './Dom';
+import { QueryStateModel } from '../Core';
+import { ComponentEvents } from '../ui/Base/Component';
+import { MODEL_EVENTS } from '../models/Model';
+
+export interface IDependentFacet {
+  reset: () => void;
+  element: HTMLElement;
+  dependsOn: string;
+  queryStateModel: QueryStateModel;
+  bind: ComponentEvents;
+}
+
+export class DependsOnManager {
+  constructor(private dependantFacet: IDependentFacet) {}
+
+  public listenToParentIfDependentFacet() {
+    if (!this.isDependentFacet) {
+      return;
+    }
+
+    this.dependantFacet.bind.onQueryState(MODEL_EVENTS.CHANGE, undefined, () => this.resetIfParentFacetHasNoSelectedValues());
+  }
+
+  public updateVisibilityBasedOnDependsOn() {
+    if (this.isDependentFacet) {
+      $$(this.dependantFacet.element).toggleClass('coveo-facet-dependent', !this.parentFacetHasSelectedValues);
```

Comment:
Maybe you don't have to do it in this PR, but you will have to come up with something better than just a adding a class. The facet component will have to be disabled to prevent some bug(s).
When it is, it will stop listening to events such as DeferredQuerySuccess. This will have the side effect of not calling this method again to reenable the component. You will have to find way to work around that. 😉

Reply:
Yeah I'll do that in another PR
Thx for the info